packages/driver/src/cy/commands/sessions/index.ts

```diff
@@ -358,11 +355,15 @@ export default function (Commands, Cypress, cy) {
           return createSession(existingSession, recreateSession)
         })
         .then(() => validateSession(existingSession))
-        .then((isValidSession: boolean) => {
+        .then(async (isValidSession: boolean) => {
           if (!isValidSession) {
             return
           }
 
+          sessionsManager.setActiveSession({ [existingSession.id]: existingSession })
+          sessionsManager.registeredSessions.set(existingSession.id, true)
+          await sessions.saveSessionData(existingSession)
```

Comment:
It looks like `sessions.saveSessionData()` is also setting `setActiveSession()`, so might be redundant

Reply:
Good call. Addressed in https://github.com/cypress-io/cypress/pull/24565/commits/ca450fe58a43c7b419c09de01842131b8cc4d7cb